Log background tasks that are slower then average (`background_schedule_pool_log.duration_threshold_milliseconds=30`)  to avoid excessive tasks logging

azat opened the pull request:
### Changelog category (leave one):
- Improvement

### Changelog entry (a [user-readable short description](https://github.com/ClickHouse/ClickHouse/blob/master/docs/changelog_entry_guidelines.md) of the changes that goes into CHANGELOG.md):
Log background tasks that are slower than average (`background_schedule_pool_log.duration_threshold_milliseconds=30`)  to avoid excessive tasks logging

Otherwise it is too much, one job in CI:
- `text_log`: 4'500'000
- `zookeeper_log`: 6'400'000
- `background_schedule_pool_log`: **54'000'000**
- `background_schedule_pool_log` where `duration_ms > 0`: **19'400**

But this is CI real values are different [1], so let's use average task
executiong time for threshold.

  [1]: https://pastila.nl/?0043db0f/96b45f5ddb995fb78c11d301d7a49f27#foCK+BCRTTH/kLTqvZ7G8g==

It should resolve flakiness due to `TOO_MANY_ROWS` [2].

  [2]: https://s3.amazonaws.com/clickhouse-test-reports/json.html?REF=master&sha=f7b1a4850b1597b443b472adbf0e2cd4f5a3b0bd&name_0=MasterCI&name_1=Stateless%20tests%20%28amd_binary%2C%20old%20analyzer%2C%20s3%20storage%2C%20DatabaseReplicated%2C%20parallel%29&name_1=Stateless%20tests%20%28amd_binary%2C%20old%20analyzer%2C%20s3%20storage%2C%20DatabaseReplicated%2C%20parallel%29

azat committed:
Increase background_schedule_pool_log.duration_threshold_milliseconds

Otherwise it is too much, one job in CI:
- text_log: 4'500'000
- zookeeper_log: 6'400'000
- background_schedule_pool_log: 54'000'000
- background_schedule_pool_log where duration_ms > 0: 19'400

But this is CI real values are different [1], so let's use average task
executiong time for threshold.

  [1]: https://pastila.nl/?0043db0f/96b45f5ddb995fb78c11d301d7a49f27#foCK+BCRTTH/kLTqvZ7G8g==

It should resolve flakiness due to TOO_MANY_ROWS [2].

  [2]: https://s3.amazonaws.com/clickhouse-test-reports/json.html?REF=master&sha=f7b1a4850b1597b443b472adbf0e2cd4f5a3b0bd&name_0=MasterCI&name_1=Stateless%20tests%20%28amd_binary%2C%20old%20analyzer%2C%20s3%20storage%2C%20DatabaseReplicated%2C%20parallel%29&name_1=Stateless%20tests%20%28amd_binary%2C%20old%20analyzer%2C%20s3%20storage%2C%20DatabaseReplicated%2C%20parallel%29

azat commented:
>It should resolve flakiness due to TOO_MANY_ROWS [1](https://s3.amazonaws.com/clickhouse-test-reports/json.html?REF=master&sha=f7b1a4850b1597b443b472adbf0e2cd4f5a3b0bd&name_0=MasterCI&name_1=Stateless%20tests%20%28amd_binary%2C%20old%20analyzer%2C%20s3%20storage%2C%20DatabaseReplicated%2C%20parallel%29&name_1=Stateless%20tests%20%28amd_binary%2C%20old%20analyzer%2C%20s3%20storage%2C%20DatabaseReplicated%2C%20parallel%29).

It does not work w/o https://github.com/ClickHouse/ClickHouse/pull/92587

Michicosun commented on programs/server/config.xml:
```diff
@@ -1260,7 +1260,7 @@
         <flush_on_crash>false</flush_on_crash>
         <!-- Only tasks longer then duration_threshold_milliseconds will be logged. Zero means log everything -->
-        <duration_threshold_milliseconds>0</duration_threshold_milliseconds>
+        <duration_threshold_milliseconds>1</duration_threshold_milliseconds>
```
This seems like a very low value, I suppose average duration will be 30-100 ms

azat committed:
tests: remove MergeTree tasks from 03746_system_background_schedule_pool_log

Those are too fast on CI and it make the test flaky after increasing the threshold

Anyway we check that those tasks presented in system.background_schedule_pool.
